lib/kredits.js

```diff
@@ -66,6 +66,23 @@ class Kredits {
     return new Kredits(provider, signer, { ipfsConfig: ipfsConfig }).init();
   }
 
+  static for (connectionOptions, kreditsOptions) {
+    let { network, rpcUrl, wallet } = connectionOptions;
+    if (!rpcUrl && network === 'local') { rpcUrl = 'http://localhost:8545'; }
+    let ethProvider, signer;
+    if (rpcUrl || network === 'local') {
+      ethProvider = new ethers.providers.JsonRpcProvider(rpcUrl);
+    } else {
+      ethProvider = new ethers.getDefaultProvider(network);
+    }
+    if (wallet) {
+      signer = wallet.connect(ethProvider);
+    } else if (ethProvider.getSigner) {
+      signer = ethProvider.getSigner();
+   }
```

Comment:
I think the linter should have caught the indentation here. It's off by one char.